Jagadish NMDC terms, without any makefile. #1089

turbomam opened the pull request:
Compiled from [Google Sheets robot template](https://docs.google.com/spreadsheets/d/1OQOtSlSjuw4OKre9YNE25-x9XnSNYaSPvbx_UjNZoBA/edit#gid=170789656)

See editor's notes for potential future improvements. I have used uppercase to indicate tasks that have been completed, like "ADDED 'composed primarily of' RO:0002473 mineral material 'ENVO:01000256'" on ENVO:03600011 'mineral horizon'

ontology ID | label
-- | --
ID | AL rdfs:label@en
ENVO:03600000 | cleanroom
ENVO:03600001 | contaminated sediment
ENVO:03600002 | cooling water
ENVO:03600003 | drinking water pipeline
ENVO:03600004 | drinking water treatment plant
ENVO:03600005 | electrical switch
ENVO:03600006 | food waste
ENVO:03600007 | formation fluid
ENVO:03600008 | intensive care unit
ENVO:03600009 | interstitial water
ENVO:03600010 | membrane bioreactor
ENVO:03600011 | mineral horizon
ENVO:03600012 | oil pipeline
ENVO:03600013 | oil sands
ENVO:03600014 | pipeline network
ENVO:03600015 | polluted river
ENVO:03600016 | saprolite
ENVO:03600017 | slurry
ENVO:03600018 | soil organic layer
ENVO:03600019 | sugarcane vinasse
ENVO:03600020 | tailings dam
ENVO:03600021 | tailings pond
ENVO:03600022 | vinasse

kaiiam commented:
@turbomam one obvious issue is the the label `oil sands` see my comment https://github.com/EnvironmentOntology/envo/issues/1089#issuecomment-825651762, where I suggest what should be a synonym and a label. Oil sands usually refers to the place not the bitumen itself, so I don't think that should be the label I think instead something like `crude bitumen` and or `bituminous sand` or a more appropriate term label should be used. `Oil sands` could be a broad synonym maybe.

kaiiam commented:
Hey @turbomam @wdduncan @pbuttigieg and I were working on this just now for our ENVO meeting. As we want to help close this soon, we've moved this along from the robot template into the protege to do a final cleanup phase. We've found working on and discussing this in a call to be very effective, perhaps we can setup a future call including you @turbomam when you're available.

kaiiam commented:
We did `cleanroom`, `contaminated sediment`, `cooling water` and `pipeline network` (see https://github.com/EnvironmentOntology/envo/issues/1111). We'll pick up again together later.

smrgeoinfo commented:
see discussion of usage of 'which' or 'that'.  https://www.grammarly.com/blog/which-vs-that/.    Seems that 'that' is more appropriate in definitions.

smrgeoinfo commented:
'mineral horizon' : "A solid layer which is formed by surface soil and subsoil, where a variety of minerals are found."

Note that 'a variety of minerals are present' in any surface soil or subsoil horizon.... So the differentia don't really differentiate.

kaiiam commented:
@smrgeoinfo it's just British vs American English, so some systems that parse/use ENVO do so expecting `which` so we've just opted to keep that as our standard. Yes other OBO projects use `that` but that's how it is.

turbomam commented:
Thanks guys. I'll be available again starting Thursday the 6th.

kaiiam commented:
Great! @turbomam let us know when you fix up the IDs and we'll merge. Afterward I think we should make a release, which I'll do.

turbomam commented:
I have assigned IRIs in the range `ENVO:036000xx` to the terms we minted yesterday evening.

Then I ran `robot report` on `envo-edit.owl` this morning and fixed an error pattern I had introduced:
- **fixed some `created_by` IRI objects lacking `http://orcid.org/`**, commit 010b31c

and two errors that preceded this work:
- **removed this empty `hasDbXref` on  `ENVO:00010505`** `AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <http://purl.obolibrary.org/obo/ENVO_00010505> <>)`, commit 4c9291e
- **removed and re-entered the label** 'concentration of nitrate in groundwater'^^xsd:string on `ENVO:09200020`, commit a6f10ff
    - robot report was saying

Level | Rule Name | Subject | Property | Value
-- | -- | -- | -- | --
ERROR | missing_label | ENVO:09200020 | rdfs:label

Then I pushed, but `continuous-integration/travis-ci/pr` is still failing. @pbuttigieg , I'm going to look at the Travis log for my own education. If you feel like this is the manifestation of a known problem with ENVO + DOSDP, then I trust you to merge.

turbomam commented:
@pbuttigieg I can't open the link you embedded in

> The failing Travis check is the mysterious artifact on DOSDP tools that we've seen before

namely https://github.com/EnvironmentOntology/envo/pull/url

I do see the failure point that you're referring to

`dosdp-tools --table-format=csv --template=./patterns/entity_attribute.yaml --outfile=modules/entity_attribute.tmp.owl --obo-prefixes=true generate --infile=modules/entity_attribute.csv`

> Unrecognized argument: --table-format=csv
Unrecognized argument: --template=./patterns/entity_attribute.yaml
Unrecognized argument: --outfile=modules/entity_attribute.tmp.owl
Unrecognized argument: --obo-prefixes=true
Makefile:543: recipe for target 'modules/entity_attribute.owl' failed

turbomam commented:
The `dosdp-tools generate` command works for me if I move the `generate` subcommand from

```
dosdp-tools \
--table-format=csv \
--template=./patterns/entity_attribute.yaml \
--outfile=modules/entity_attribute.tmp.owl \
--obo-prefixes=true \
generate \
--infile=modules/entity_attribute.csv
```

to 

```
dosdp-tools \
generate \
--table-format=csv \
--template=./patterns/entity_attribute.yaml \
--outfile=modules/entity_attribute.tmp.owl \
--obo-prefixes=true \
--infile=modules/entity_attribute.csv
```

I'm installing some of the other dependencies from the Makefile now and will try applying that change to the Makefile.

kaiiam commented:
Interesting @matentzn can you comment on the DOSDP issue, especially since @turbomam you didn't use or call it in this PR correct?

turbomam commented:
Thanks, @kaiiam I just want to make sure we understand each other. The error travis reports come from `dosdp generate` as invoked by the Makefile.

```dosdp-tools \
--table-format=csv \
--template=./patterns/entity_attribute.yaml \
--outfile=modules/entity_attribute.tmp.owl \
--obo-prefixes=true \
generate \
--infile=modules/entity_attribute.csv``

I was able to get it working on my laptop, by manually entering

```
dosdp-tools \
generate \
--table-format=csv \
--template=./patterns/entity_attribute.yaml \
--outfile=modules/entity_attribute.tmp.owl \
--obo-prefixes=true \
--infile=modules/entity_attribute.csv
```

Now I'm trying to get the same thing to run as part of the Makefile.

turbomam commented:
@pbuttigieg @kaiiam @cmungall @wdduncan Wow, thanks for all of you help with this!

- The terms required by Jagadish from NMDC (and new intermediate classes where required) are all in the `envo-edit.owl` from the `Jagadish-NMDC-terms-no-makefile` branch now.
- Those terms all have standard IRIs, eg `http://purl.obolibrary.org/obo/ENVO_036000xx`
- I don't see any errors when running `robot report` locally
- I made a sight reordering of the Makefile so that there's no `dosdp-tools generate` error any more

@pbuttigieg let me know if I should provide any more information before you merge.
Then @kaiiam , you'll make a release when you get a chance, right?

kaiiam commented:
Great thanks @turbomam. I'll review it now. Seems the [changes to the makefile](https://github.com/EnvironmentOntology/envo/pull/1097/files/a6f10ffa2c05e9ac81ef836c87aec18fea2d35c0..c1d3d66e543f79ed25a55a39f0d0c4a9f03b4fa3) fixed the issue hopefully it doesn't cause any issues downstream. I presume this wouldn't @matentzn? 

If this is settled and were ok to go ahead with a release I'll aim to make one early next week.

matentzn commented:
Haha if this fixes the issue I will propose a toast to @turbomam from my balcony. Change is fine.

kaiiam reviewed (approved):
I just went through it again and added a few more commits mostly to consolidate down to only having one editor note per term, and adding a bit more to the editor notes including xrefs. This looks good to me. Approving.

pbuttigieg commented:
Looks good all - thanks @turbomam for your investigation

cmungall commented:
I disagree with the grammarly article. "Which" is absolutely fine from a
grammar perspective. And IMO it sounds better.

Unfortunately many people in the US reflexively think many appropriate
usage of "which" is wrong, due to the influence of grammar books written in
the 50s that had plain bad advice

As a result, 'that' is more common. All other ontologies use 'that' in the
genus-differentia definition.

For this reason, I think ENVO should switch to 'that' for consistency
across OBO. Just as we standardise^H^Hze on American spelling. It makes me
sad, but consistency is good for users, and helps us write automatic style
checking tools etc

On Mon, May 3, 2021 at 9:14 AM Stephen Richard ***@***.***>
wrote:

> see discussion of usage of 'which' or 'that'.
> https://www.grammarly.com/blog/which-vs-that/. Seems that 'that' is more
> appropriate in definitions.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/EnvironmentOntology/envo/pull/1097#issuecomment-831368548>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAMMOMVJGJFTYE7CNFYR2LTL3DUXANCNFSM43VEHCVQ>
> .
>

pbuttigieg reviewed (approved):
Good for now - many thanks
